trie-db/Cargo.toml

```
@@ -12,6 +12,7 @@ rand = { version = "0.6", default-features = false }
 smallvec = "1.0.0"
 hash-db = { path = "../hash-db", default-features = false, version = "0.15.2"}
 hashbrown = { version = "0.6.3", default-features = false }
+rustc-hex = { version = "2.1.0", optional = true }
```

Does it use actually `std` feature of `rust-hex`? I thought it's only required for [this](https://github.com/debris/rustc-hex/blob/7c8e13d68db1253f9a5e9a6e9160ad913e249153/src/lib.rs#L138)? `fmt` is available in `no_std` as well.

---

The whole tracing is done only in `std`, so I didn't change any behavior here.

---

I see, we still can use `default-features = false`, right?
```suggestion
rustc-hex = { version = "2.1.0", default-features = false, optional = true }
```

---

Yes, I guess we can, but could you elaborate on what we gain here? (or is it just because we don't need to have that features, so why enable them in the first place?)

---

I guess it would enable us to transition to `no-std` tracing easier (if we consider that path) as well as less code to compile in `std` mode.